src/ray/core_worker/task_event_buffer.h

```diff
@@ -273,9 +275,60 @@ class TaskEventBufferImpl : public TaskEventBuffer {
   const std::string DebugString() override;
 
  private:
+  /// Add a task status event to be reported.
+  ///
+  /// \param status_event Task status event.
+  void AddTaskStatusEvent(std::unique_ptr<TaskEvent> status_event) LOCKS_EXCLUDED(mutex_);
+
+  /// Add a task profile event to be reported.
+  ///
+  /// \param profile_event Task profile event.
+  void AddTaskProfileEvent(std::unique_ptr<TaskEvent> profile_event)
+      LOCKS_EXCLUDED(profile_mutex_);
+
+  /// Get data related to task status events to be send to GCS.
+  ///
+  /// \param[out] status_events_to_send Task status events to be sent.
+  /// \param[out] dropped_task_attempts_to_send Task attempts that were dropped due to
+  ///             status events being dropped.
+  void GetTaskStatusEventsToSend(
+      std::vector<std::unique_ptr<TaskEvent>> *status_events_to_send,
+      absl::flat_hash_set<TaskAttempt> *dropped_task_attempts_to_send)
+      LOCKS_EXCLUDED(mutex_);
+
+  /// Get data related to task profile events to be send to GCS.
+  ///
+  /// \param[out] profile_events_to_send Task profile events to be sent.
+  /// \param dropped_task_attempts_to_send Task attempts that were dropped due to
+  ///        status events being dropped. We will not send any profile events for these
+  ///        task attempts.
+  /// \param status_events_to_send Task status events that will be sent to GCS. We will
+  ///        prioritize sending profile events for these task attempts first.
```

Comment:
is this necessary requirements? I feel like this method is too complicated due to this featuare. Why don't we just send profile events FIFO? (IIUC this is the status quo?)

Reply:
I assume it is the optimization, but is it necessary? (feel like GCS can just dedup eventually as it knows all dropped task attempts?)

Reply:
> Why don't we just send profile events FIFO? (IIUC this is the status quo?)

Yeah, we could do this.